packages/report-generator-runner/src/runner/runner.ts

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { inject, injectable } from 'inversify';
+import { GlobalLogger } from 'logger';
+import { PrivacyScanResult } from 'scanner-global-library';
+import { OnDemandPageScanRunResultProvider, WebsiteScanResultProvider, ReportWriter, GeneratedReport } from 'service-library';
+import { OnDemandPageScanReport, OnDemandPageScanResult, OnDemandPageScanRunState, ScanError, WebsiteScanResult } from 'storage-documents';
+import { System, ServiceConfiguration } from 'common';
+import _ from 'lodash';
+import { RunMetadataConfig } from '../run-metadata-config';
+import { ReportGeneratorRunnerTelemetryManager } from '../report-generator-runner-telemetry-manager';
+import { ReportGeneratorMetadata } from '../types/report-generator-metadata';
+
+@injectable()
+export class Runner {
```

Comment:
Are there unit tests for this file?

Reply:
No, I'll change the logic next pr and add tests